jetstream/dryrun.py

```diff
@@ -35,10 +37,28 @@ def __init__(self, error: Any, sql: str):
 def dry_run_query(sql: str) -> None:
     """Dry run the provided SQL query."""
     try:
+        auth_req = GoogleAuthRequest()
+        creds, _ = google.auth.default(scopes=["https://www.googleapis.com/auth/cloud-platform"])
+        creds.refresh(auth_req)
+        if hasattr(creds, "id_token"):
+            # Get token from default credentials for the
+            # current environment created via Cloud SDK run
+            id_token = creds.id_token
+        else:
+            # If the environment variable GOOGLE_APPLICATION_CREDENTIALS
+            # is set to service account JSON file,
+            # then ID token is acquired using this service account credentials.
+            id_token = fetch_id_token(auth_req, DRY_RUN_URL)
+
         r = requests.post(
             DRY_RUN_URL,
-            headers={"Content-Type": "application/json"},
-            data=json.dumps({"dataset": "mozanalysis", "query": sql}).encode("utf8"),
+            headers={
+                "Content-Type": "application/json",
+                "Authorization": f"Bearer {id_token}",
+            },
+            data=json.dumps(
+                {"dataset": "mozanalysis", "project": "moz-fx-data-experiments", "query": sql}
+            ).encode("utf8"),
```

Comment:
Not sure why I'm just thinking of this now, but is there a reason to use a CloudRun function instead of just using the python client to create a [dry_run query](https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.job.QueryJobConfig#google_cloud_bigquery_job_QueryJobConfig_dry_run)?

Reply:
Ooh yea now I remember discussing or reading this sometime, thanks!